lodash/lodash.d.ts

```diff
@@ -263,12 +263,12 @@ declare module _ {
         * @param array Array to compact.
         * @return (Array) Returns a new array of filtered values.
         **/
-        compact<T>(array: Array<T>): T[];
+        compact<T>(array?: Array<T>): T[];
```

Comment:
I think type should lead a code to valid style. and type should not block writing code.
I'll merge this PR if your code is useful in real world.